### ISSUE TITLE
finetune tokenizer时loss为负的问题

### ISSUE THREAD

**Issue by 13922740870** [shiyu-coder/Kronos#81]
我在用期货数据做finetune时，tokenizer的训练过程一开始的Loss就为负数

Using device: cuda
Loading weights from local directory
Model Size: 4.0M
BATCHSIZE: 50
Effective total batch size: 50
Creating dataloaders...
[TRAIN] Pre-computing sample indices...
[TRAIN] Found 165400 possible samples. Using 100000 per epoch.
[VAL] Pre-computing sample indices...
[VAL] Found 49156 possible samples. Using 20000 per epoch.
Train dataset size: 100000, Validation dataset size: 20000
Dataloaders created. Train steps/epoch: 2000, Val steps: 400
[Epoch 1/30, Step 100/2000] LR 0.000021, Loss: -0.0235
[Epoch 1/30, Step 200/2000] LR 0.000025, Loss: -0.0235
[Epoch 1/30, Step 300/2000] LR 0.000032, Loss: -0.0244
[Epoch 1/30, Step 400/2000] LR 0.000041, Loss: -0.0252
[Epoch 1/30, Step 500/2000] LR 0.000052, Loss: -0.0231
[Epoch 1/30, Step 600/2000] LR 0.000065, Loss: -0.0233
[Epoch 1/30, Step 700/2000] LR 0.000079, Loss: -0.0246
[Epoch 1/30, Step 800/2000] LR 0.000094, Loss: -0.0222
[Epoch 1/30, Step 900/2000] LR 0.000110, Loss: -0.0241
[Epoch 1/30, Step 1000/2000] LR 0.000126, Loss: -0.0228
[Epoch 1/30, Step 1100/2000] LR 0.000141, Loss: -0.0245

如果我把tokenizer的参数做下修改，Loss是正的。
model.tokenizer.bsq.gamma = 0.10

麻烦请问正确的解决方式

**Comment by shiyu-coder** [comment]
tokenizer的loss为负是正常的，不影响训练，具体逻辑可以参考BSQ的原论文

**Comment by 13922740870** [comment]
明白了，谢谢！
